### ISSUE TITLE
Installation on Mac 

### ISSUE THREAD

**Issue by minouHub** [radis/radis#696]
### 🐛 Describe the bug

Two of my colleagues could not install radis on Mac with either pip or conda. 

The culprit is clearly `nvidia-cufft-cu11`. @dcmvdbekerom, I guess this package could be optional after #624?
**User 1:**
```
Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Collecting radis
  Downloading radis-0.15.1-py3-none-any.whl.metadata (13 kB)
Requirement already satisfied: astropy>=4.3.1 in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (6.1.0)
Collecting astroquery>=0.4.6 (from radis)
  Downloading astroquery-0.4.7-py3-none-any.whl.metadata (7.2 kB)
Requirement already satisfied: beautifulsoup4>=4.10.0 in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (4.12.3)
Collecting cantera>=2.5.1 (from radis)
  Downloading Cantera-3.0.1-cp310-cp310-macosx_11_0_arm64.whl.metadata (7.8 kB)
Collecting configparser (from radis)
  Downloading configparser-7.1.0-py3-none-any.whl.metadata (5.4 kB)
Collecting habanero>=1.2.0 (from radis)
  Downloading habanero-1.2.6-py2.py3-none-any.whl.metadata (14 kB)
Requirement already satisfied: h5py>=3.2.1 in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (3.11.0)
Requirement already satisfied: joblib in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (1.4.2)
Collecting lmfit (from radis)
  Downloading lmfit-1.3.2-py3-none-any.whl.metadata (13 kB)
Requirement already satisfied: matplotlib in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (3.9.2)
Requirement already satisfied: numpy in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (1.26.4)
Requirement already satisfied: numba in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (0.60.0)
Requirement already satisfied: pandas in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (2.1.4)
Requirement already satisfied: plotly>=2.5.1 in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (5.22.0)
Requirement already satisfied: psutil in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (5.9.0)
Requirement already satisfied: tables in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (3.10.1)
Requirement already satisfied: scipy>=1.4.0 in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (1.13.1)
Requirement already satisfied: seaborn in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (0.13.2)
Collecting termcolor (from radis)
  Downloading termcolor-2.4.0-py3-none-any.whl.metadata (6.1 kB)
Collecting specutils (from radis)
  Downloading specutils-1.16.0-py3-none-any.whl.metadata (6.0 kB)
Requirement already satisfied: lxml in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (4.9.1)
Collecting hjson (from radis)
  Downloading hjson-3.1.0-py3-none-any.whl.metadata (2.6 kB)
Collecting publib (from radis)
  Downloading publib-0.4.0-py3-none-any.whl.metadata (1.0 kB)
Collecting hitran-api (from radis)
  Downloading hitran_api-1.2.2.2-py2.py3-none-any.whl.metadata (172 bytes)
Collecting peakutils (from radis)
  Downloading PeakUtils-1.3.5-py3-none-any.whl.metadata (1.6 kB)
Requirement already satisfied: ruamel.yaml in /opt/homebrew/anaconda3/lib/python3.10/site-packages (from radis) (0.17.21)
Collecting json-tricks>=3.15.0 (from radis)
  Downloading json_tricks-3.17.3-py2.py3-none-any.whl.metadata (16 kB)
Collecting mpldatacursor (from radis)
  Downloading mpldatacursor-0.7.1-py3-none-any.whl.metadata (8.8 kB)
Collecting nvidia-cufft-cu11 (from radis)
  Downloading https://developer.download.nvidia.com/compute/redist/nvidia-cufft-cu11/nvidia-cufft-cu11-2022.4.8.tar.gz (16 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/yd/7v5rljn11td1hr16pjmvg4jm0000gr/T/pip-install-cvglf315/nvidia-cufft-cu11_752dd024514b4dde9dc0c78aeb212348/setup.py", line 40, in <module>
          raise OSError("Your operating system is not supported by "
      OSError: Your operating system is not supported by `nvidia-cufft-cu11`: darwin.
      Only these systems are supported: `['linux', 'win32']`.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

```

**User 2:**
```
(Radis) guires@wifibridge ~ % pip install radis
Collecting radis
  Downloading radis-0.15.1-py3-none-any.whl.metadata (13 kB)
Collecting astropy>=4.3.1 (from radis)
  Downloading astropy-6.1.3-cp310-cp310-macosx_11_0_arm64.whl.metadata (10 kB)
Collecting astroquery>=0.4.6 (from radis)
  Downloading astroquery-0.4.7-py3-none-any.whl.metadata (7.2 kB)
Collecting beautifulsoup4>=4.10.0 (from radis)
  Downloading beautifulsoup4-4.12.3-py3-none-any.whl.metadata (3.8 kB)
Collecting cantera>=2.5.1 (from radis)
  Downloading Cantera-3.0.1-cp310-cp310-macosx_11_0_arm64.whl.metadata (7.8 kB)
Collecting configparser (from radis)
  Downloading configparser-7.1.0-py3-none-any.whl.metadata (5.4 kB)
Collecting habanero>=1.2.0 (from radis)
  Downloading habanero-1.2.6-py2.py3-none-any.whl.metadata (14 kB)
Collecting h5py>=3.2.1 (from radis)
  Downloading h5py-3.11.0-cp310-cp310-macosx_11_0_arm64.whl.metadata (2.5 kB)
Collecting joblib (from radis)
  Downloading joblib-1.4.2-py3-none-any.whl.metadata (5.4 kB)
Collecting lmfit (from radis)
  Downloading lmfit-1.3.2-py3-none-any.whl.metadata (13 kB)
Collecting matplotlib (from radis)
  Downloading matplotlib-3.9.2-cp310-cp310-macosx_11_0_arm64.whl.metadata (11 kB)
Collecting numpy (from radis)
  Downloading numpy-2.1.1-cp310-cp310-macosx_14_0_arm64.whl.metadata (60 kB)
Collecting numba (from radis)
  Downloading numba-0.60.0-cp310-cp310-macosx_11_0_arm64.whl.metadata (2.7 kB)
Collecting pandas (from radis)
  Downloading pandas-2.2.2-cp310-cp310-macosx_11_0_arm64.whl.metadata (19 kB)
Collecting plotly>=2.5.1 (from radis)
  Downloading plotly-5.24.1-py3-none-any.whl.metadata (7.3 kB)
Collecting psutil (from radis)
  Downloading psutil-6.0.0-cp38-abi3-macosx_11_0_arm64.whl.metadata (21 kB)
Collecting tables (from radis)
  Downloading tables-3.10.1-cp310-cp310-macosx_11_0_arm64.whl.metadata (2.2 kB)
Collecting scipy>=1.4.0 (from radis)
  Downloading scipy-1.14.1-cp310-cp310-macosx_14_0_arm64.whl.metadata (60 kB)
Collecting seaborn (from radis)
  Downloading seaborn-0.13.2-py3-none-any.whl.metadata (5.4 kB)
Collecting termcolor (from radis)
  Downloading termcolor-2.4.0-py3-none-any.whl.metadata (6.1 kB)
Collecting specutils (from radis)
  Downloading specutils-1.16.0-py3-none-any.whl.metadata (6.0 kB)
Collecting lxml (from radis)
  Downloading lxml-5.3.0-cp310-cp310-macosx_10_9_universal2.whl.metadata (3.8 kB)
Collecting hjson (from radis)
  Downloading hjson-3.1.0-py3-none-any.whl.metadata (2.6 kB)
Collecting publib (from radis)
  Downloading publib-0.4.0-py3-none-any.whl.metadata (1.0 kB)
Collecting hitran-api (from radis)
  Downloading hitran_api-1.2.2.2-py2.py3-none-any.whl.metadata (172 bytes)
Collecting peakutils (from radis)
  Downloading PeakUtils-1.3.5-py3-none-any.whl.metadata (1.6 kB)
Collecting ruamel.yaml (from radis)
  Downloading ruamel.yaml-0.18.6-py3-none-any.whl.metadata (23 kB)
Collecting json-tricks>=3.15.0 (from radis)
  Downloading json_tricks-3.17.3-py2.py3-none-any.whl.metadata (16 kB)
Collecting mpldatacursor (from radis)
  Downloading mpldatacursor-0.7.1-py3-none-any.whl.metadata (8.8 kB)
Collecting nvidia-cufft-cu11 (from radis)
  Downloading nvidia-cufft-cu11-0.0.1.dev5.tar.gz (8.1 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/y6/cq5bwz1j4t1drf5w3zshr1dw0000gq/T/pip-install-c389pnee/nvidia-cufft-cu11_d6d1df
4fc89f44d6b90218abf91e9ed0/setup.py", line 150, in <module>
          raise RuntimeError(open("ERROR.txt", "r").read())
      RuntimeError:
      ###########################################################################################
      The package you are trying to install is only a placeholder project on PyPI.org repository.
      This package is hosted on NVIDIA Python Package Index.
      
      This package can be installed as:
      ```
      $ pip install nvidia-pyindex
      $ pip install nvidia-cufft-cu11
      ```
      ###########################################################################################
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
× Encountered error while generating package metadata.
╰─> See above for output.
note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

### 💡 Possible solutions

- [x] Implement install test on Mac
- [x] Make nvidia-cufft-cu11 optional or find a version compatible with Mac

### 🎲 Radis version

0.15.1

### 💻 Operating system

MacOS

**Comment by dcmvdbekerom** [comment]
Yes we should make the package optional

However, some time ago we decided to not support MacOS for gpu stuff because nobody was using it. I'm inclined to stick with that because I have no access to a MacOS system

**Comment by minouHub** [comment]
A colleague working on the impact of radiation in flames and using Mac was willing to use radis, so it's not nobody anymore. I agree with you that it should still be a low priority and should not prevent the merge of a new feature for Linux and Windows users. I'll try to sort this specific problem with a minor fix

**Comment by dcmvdbekerom** [comment]
Just to be clear, I completely agree that MacOS users should be able to use "vanilla" radis, so we should make that GPU part optional. This would help with reducing the total download size as well.

MacOS support is not too much of a stretch as long as we have someone in the developers team that has access to MacOS. Not just for testing, but particularly for compilation of dynamic libraries, which will be required once we move to Vulkan.

**Comment by akyriako** [comment]
Hi there, I have opened the PR #699 which could eliminate all installation problems (on any OS) as it introduces a standardized, common defined, container image to work with in VSCode. No additional installation, except docker, podman, orbstack or any other similar technology that allows the host OS to run containers, would be required on the host OS.
